rgbd/test/test_server_templ.h

```diff
@@ -0,0 +1,75 @@
+#ifndef TEST_SERVER_TEMPL_H_
```

Comment:
As a general comment on your tests (which I didn't inspect your tests in detail): I'm under the impression that you mainly test the 'happy flow'. Do you also test switching between different modes on purpose? Since that's the reason for this exercise, that's quite essential. 

In any case, adding a couple of comments to your tests would make it a lot more insightful which cases your actually testing.